pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -48,6 +48,17 @@ self: super: {
   # Needs older QuickCheck version
   attoparsec-varword = dontCheck super.attoparsec-varword;
 
+  ShellCheck = overrideCabal super.ShellCheck (drv: {
+    patches = [
+      # cabal 3.0 support
+      ( pkgs.fetchpatch {
+        url = "https://github.com/koalaman/shellcheck/commit/2c026f1ec7c205c731ff2a0ccd85365f37245.patch";
+        sha256 = "0z6yf350ngr6rwfkvdy670c476fgzj8a0n4ppdm1xr8r1lij7sfz";
+        excludes = [ "Dockerfile" ];
```

Comment:
Note that `fetchpatch` cannot remove or create files (thus, `Setup.hs` still exists and needs to be removed in `postPatch`).

I'm only parroting what I was told on #nixos-dev, but changing to `excludes` in my local patch and commenting the removal of `Setup.hs` causes the build to continue to fail.

Reply:
even so, it appears that removing the contents of the file is sufficient enough

Reply:
I forgot to update the `sha256` :laughing:

Reply:
fetchpatch appears to do the normalization before finalizing the the diff. Which means that if you pass options to it, you will need to re-determine the final output sha. this is what i meant in https://github.com/NixOS/nixpkgs/pull/81334#issuecomment-592779934

Reply:
otherwise you're actually picking up the old "cached" one

Reply:
Yeah, oops. Builds fine.

```
[3 built]
https://github.com/NixOS/nixpkgs/pull/81334
1 package marked as broken and skipped:
shellinabox

2 package built:
nix-info-tested shellcheck
```